planning/behavior_path_planner/src/scene_module/pull_over/util.cpp

```diff
@@ -344,6 +68,30 @@ lanelet::ConstLanelets getPullOverLanes(const RouteHandler & route_handler)
   return pull_over_lanes;
 }
 
+bool hasEnoughDistanceToParkingStart(
+  const PathWithLaneId & path, const Pose & current_pose, const Pose & start_pose,
+  const double current_vel, const double maximum_deceleration, const double decide_path_distance)
+{
+  const auto dist_to_start_pose = calcSignedArcLength(
+    path.points, current_pose, start_pose.position, std::numeric_limits<double>::max(), M_PI_2);
```

Comment:
calculate ego_seg_idx here with findFirstNearestSegmentIndexWithSoftConstraints, and use calcSignedArcLength which have segment_idx in arguments.

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/1911/commits/b47d458c9870cfcb74d01d9866159bd273ee1972

Reply:
I should've mentioned that we have `findEgoSegmentIndex` in scene module interface. :pray:

Reply:
but this is the function in utils, `findEgoSegmentIndex` does not seems to be available.

Reply:
Ah I got it. Thanks.